### PR TITLE
WIP: Region discoveries

### PR DESCRIPTION
**Pull request by cjab**
Still quite a bit of work to do here but some progress on the 0x22 Region fragment.

### CODE REVIEW

**Comment by nickgal** [comment]
awesome stuff! :)

**Comment by bryab** [comment]
the mysteries of the ancients shall soon be revealed to all

File: crates/libeq_wld/src/parser/fragments/bsp_region.rs [hunk 1]
```diff
@@ -374,21 +364,20 @@ impl BspRegionFragmentData4Entry {
...
-pub struct BspRegionFragmentData5Entry {
-    /// _Unknown_ - Usually 0.
-    params1: (u32, u32, u32),
-
-    /// _Unknown_ - Usually 0.
-    params2: u32,
+pub struct Obstacle {
+    /// bit 0 - is a FLOOR
+    /// bit 1 - is a GEOMETRYCUTTINGOBSTACLE
+    /// bit w - has USERDATA %s
```
Comment (nickgal) [inline crates/libeq_wld/src/parser/fragments/bsp_region.rs]: typo? should be bit 2?

Comment (cjab) [reply crates/libeq_wld/src/parser/fragments/bsp_region.rs]: Yep, saw that last night 😄. I just pushed a fix and a bit more info.

**Comment by cjab** [comment]
I haven't tried this on a _huge_ number of zones but it seems reasonably correct 😄. I have a feeling a number of these fields are mostly or entirely unused in EQ.

**Pull request merged**